lib/router.js

```diff
@@ -79,9 +79,10 @@ Cherrytree.prototype.map = function (routes) {
 
   function eachBranch (node, memo, fn) {
     node.routes.forEach(function (route) {
-      if (!route.routes || route.routes.length === 0) {
+      if (!route.options || route.options.abstract !== true) {
```

Comment:
Also users should always have a catch handler middleware so users don't end up stuck somewhere, and that or similar middleware can redirect for what would be the abstract non-leaf nodes.